### DISCUSSION TITLE
does not seem to work on ubuntu 20.10 and 20.04.1

### DISCUSSION THREAD

**Discussion by lxwang42** [danielewood/sierra-wireless-modems#58] [General] [answered]
with dw5811e, does not seem to be able to complete the whole process on ubuntu does not seem to work on ubuntu 20.10.
but it works smoothly on ubuntu 18.04. 
here is my log:

Waiting for modem to reboot...
Downloading https://source.sierrawireless.com/~/media/support_downloads/airprime/74xx/fw/7455/SWI9X30C_02.33.03.00_GENERIC_002.072_000.zip
% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
100 60.7M 100 60.7M 0 0 609k 0 0:01:42 0:01:42 --:--:-- 599k
Archive: SWI9X30C_02.33.03.00_GENERIC_002.072_000.zip
inflating: SWI9X30C_02.33.03.00.cwe
inflating: SWI9X30C_02.33.03.00_GENERIC_002.072_000.nvu

\-\-\-

Flashing SWI9X30C_02.33.03.00.cwe onto Generic Sierra Modem...
loading device information before the update...
error: couldn't open QMI device: Failure
Firmware Update failed, exiting...

**Comment by lxwang42** [comment]
update: tested on Raspberry Pi OS Linux kernel 5.4.79 with em7455, the script works fine.

**Reply by danielewood** [reply]
Yes, I ran into a similar issue a few days ago, the fix was to set the modem to QMI mode(both swi util and AT commands), then flash it.

I suspect there is a bug that needs filed upstream with libqmi.

**Comment by danielewood** [comment, marked as answer]
As a workaround, I've switched to always flashing in boothold mode in #59

**Reply by techguy121400** [reply]
Hello I recently tried to update my MC7455 modem but I think the Server isn't working anymore can you confirm ?

**Reply by ghost** [reply]
@danielewood

source.sierrawireless.com is no longer functional. Do you happen to have the zip saved? I believe if there is a copyright issue than the corresponding license can be uploaded as well.

**Comment by lxwang42** [comment]
According to Sierra’s news release, Sierra’s IT systems are experiencing
ransom attacks, and they are working on it. Hopefully their services will
resume soon, otherwise you can always email them for the firmware update.
Regards

**Reply by ghost** [reply]
Ah, that clears up the website suddenly going down. Thanks for letting us know.
